graphql_compiler/compiler/compiler_entities.py

```diff
@@ -66,23 +68,26 @@ def __eq__(self, other):
 
     # pylint: enable=protected-access
 
-    def __ne__(self, other):
+    def __ne__(self, other: Any) -> bool:
         """Check another object for non-equality against this one."""
         return not self.__eq__(other)
 
-    @abstractmethod
-    def to_gremlin(self):
+    def to_gremlin(self) -> str:
         """Return the Gremlin unicode string representation of this object."""
         raise NotImplementedError()
 
 
+AliasType = Union[Alias, CTE]
+AliasesDictType = Dict[Tuple[Tuple[str, ...], Optional[Tuple[Tuple[str, str], ...]],], AliasType]
```

Comment:
Suggestion: consider adding a brief description of what these are meant to represent at a high level.

Reply:
I'd love to — but I actually don't know if they are even correct, let alone how to best describe them. @bojanserafimov, help? 😄

Reply:
To avoid playing a game of telephone, this might be a good fit for a separate PR by @bojanserafimov.

Reply:
Gotcha - maybe add a TODO/issue? Because right now I'm not sure what I'm looking at :)...

Reply:
Opened #711 for it.